packages/turf-transform-rotate/index.js

```diff
@@ -0,0 +1,115 @@
+var helpers = require('@turf/helpers');
+var centroid = require('@turf/centroid');
+var invariant = require('@turf/invariant');
+var rhumbBearing = require('@turf/rhumb-bearing');
+var rhumbDistance = require('@turf/rhumb-distance');
+var rhumbDestination = require('@turf/rhumb-destination');
+var getCoord = invariant.getCoord;
+var getCoords = invariant.getCoords;
+var point = helpers.point;
+var feature = helpers.feature;
+var polygon = helpers.polygon;
+var multiPoint = helpers.multiPoint;
+var lineString = helpers.lineString;
+var multiPolygon = helpers.multiPolygon;
+var multiLineString = helpers.multiLineString;
+
+/**
+ * Rotates any geojson Feature or Geometry of a specified angle, around its `centroid` or a given `pivot` point;
+ * all rotations follow the right-hand rule: https://en.wikipedia.org/wiki/Right-hand_rule
+ *
+ * @name rotate
+ * @param {Geometry|Feature<any>} geojson object to be rotated
+ * @param {number} angle of rotation (along the vertical axis), from North in decimal degrees, negative clockwise
+ * @param {Geometry|Feature<Point>|Array<number>} [pivot=`centroid`] point around which the rotation will be performed
+ * @param {number} [xRotation=0] angle of rotation, respect to the horizontal (x,y) plain, about the x axis (not implemented);
```

Comment:
@stebogit Just some random thoughts. It states here that x and y rotations are not yet implemented. This will require some careful thought on how to combine with the given rhumb approach for a few reasons:
 - I don't know of a natural geospatial way to rotate along those axes.
 - The rhumb approach is not really a z rotation but a quasi-z rotation that rotates the points in x, y, and z along the earth surface.
 - Most importantly, the order of rotations in 3D are not interchangeable. So, just providing a scalar value for `xRotation` and `yRotation` would not be enough to define a unique rotation.